test/runtest.py

```diff
@@ -1205,6 +1205,65 @@ def test_multi_threading_in_worker():
         # test multi-threading in the worker
         ray.get(test_multi_threading_in_worker.remote())
 
+    @unittest.skipIf(
+        os.environ.get("RAY_USE_XRAY") != "1",
+        "This test only works with xray.")
+    def testFreeObjectsMultiNode(self):
+        ray.worker._init(
+            start_ray_local=True,
+            num_local_schedulers=3,
+            num_workers=1,
+            num_cpus=[100, 5, 10],
+            num_gpus=[0, 5, 1],
+            use_raylet=True)
+
+        @ray.remote(num_cpus=11)
+        def run_on_0():
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        @ray.remote(num_gpus=2)
+        def run_on_1():
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        @ray.remote(num_cpus=6, num_gpus=1)
+        def run_on_2():
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        def create():
+            c = run_on_0.remote()
+            a = run_on_1.remote()
+            b = run_on_2.remote()
+            (l1, l2) = ray.wait([a, b, c], num_returns=3)
+            assert len(l1) == 3
+            assert len(l2) == 0
+            return (a, b, c)
+
+        def flush():
+            # Flush the Release History.
+            # Current Plasma Client Cache will maintain 64-item list.
+            # If the number changed, this will fail.
+            print("Start Flush!")
+            for i in range(1600):
```

Comment:
Maybe make this smaller? E.g., just 64 instead of 1600. We want to make sure it is getting removed from the call to `free` and not accidentally evicted.